### DISCUSSION TITLE
Function Refused By CPU

### DISCUSSION THREAD

**Discussion by GleeisonRodrigues** [gijzelaerr/python-snap7#455] [Q&A] [answered]
Having an issue with try to read a DB. My connection is successful but I get the rejection on the read function. In the DB example I have created a global DB26 and I am trying to read DB26.00. The db optimization is turned off as per the instruction and also I enable the features ALLOW PUT/GET .

CPU is 6ES7 214-1HG40-0XB0 CPU 1214C DC/DC/Rly Firmware Version 4.2

python version 3.11.4

Reading Example:

import snap7.client as c
from snap7.util import *
from snap7.types import *

if __name__ == "__main__":
    plc = c.Client()
    plc.connect('192.168.0.20',0,1)
    print("Connection Sucessful")
    data = plc.db_read(26,0,1)

![image](https://github.com/gijzelaerr/python-snap7/assets/18513746/5fcf864c-757e-4225-a4e7-aa86c70e00cf)

![image](https://github.com/gijzelaerr/python-snap7/assets/18513746/2e21079b-59b7-4db3-bca3-3c4f692619e9)

![image](https://github.com/gijzelaerr/python-snap7/assets/18513746/00f5c304-e506-4fe7-bd02-524d51c68e69)

**Comment by nikteliy** [comment]
Hi @GleeisonRodrigues
Check the properties of your DB block. The `optimized block access` checkbox should be unchecked.

![1](https://github.com/gijzelaerr/python-snap7/assets/52915342/3a4c7513-a70d-4499-a3c7-d69ba3e9e1ab)

![2](https://github.com/gijzelaerr/python-snap7/assets/52915342/36563a40-3b1c-40cd-9c60-9e8956cff041)

**Comment by GleeisonRodrigues** [comment]
Hi @nikteliy , I did that.
![image](https://github.com/gijzelaerr/python-snap7/assets/18513746/29c137d6-3a12-45a3-bd2b-e0362c633882)

**Comment by nikteliy** [comment]
Did you reinitialize the db after that?
Have you uploaded the hardware configuration to the PLC after making changes?
It works for me, but i have firmware 4.0.
![output](https://github.com/gijzelaerr/python-snap7/assets/52915342/fa1e8865-5103-441e-a9b7-c1c6dcc98380)

**Comment by GleeisonRodrigues** [comment]
Hi I also did that :(

**Reply by nikteliy** [reply]
I've run out of ideas. Can you post here the project archive, so I can test it on my PLC? I only need the hardware config and that DB.

**Comment by GleeisonRodrigues** [comment]
[CENTRAL_BALANCE.zip](https://github.com/gijzelaerr/python-snap7/files/12263909/CENTRAL_BALANCE.zip)

Hi @nikteliy , here is my project. thanks for your help.

**Reply by nikteliy** [reply]
Sorry, I cannot open it. I only have TIA Portal v15.1, and older versions don't work with projects created in a newer version.

**Comment by gijzelaerr** [comment, marked as answer]
Closing as stale. This was an S7-1200 configuration issue. Uncheck 'optimized block access' and reinitialize the DB after the change.
